src/schematools/types.py

```diff
@@ -1031,27 +1055,25 @@ def db_name(
             version_postfix = self.version.signif
         if with_dataset_prefix:
             dataset_prefix = to_snake_case(self.dataset.id)
+
+        shortname = to_snake_case(self.shortname)
         if self.nested_table or self.through_table:
             # We don't automatically shorten user defined table names. Automatically generated
             # names, however, should be shortened as the user has no direct control over them.
-            db_table_name = "_".join(filter(None, (dataset_prefix, to_snake_case(self.name))))
-            additional_underscores = len(list(filter(None, (version_postfix, postfix))))
+            db_table_name = _name_join(dataset_prefix, shortname)
+            additional_underscores = _name_join_count(dataset_prefix, shortname)
```

Comment:
This is incorrect:

```python
>>> _name_join("foo", "bar")
'foo_bar'
>>> _name_join_count("foo", "bar")
2
```

... but there's only one underscore. (Not introduced by this PR.)